### PR TITLE
Add position restraints for Site

### PR DESCRIPTION
**Pull request by daico007**
Add `position_restraints` for site and write it out in GROMACS top writer. Pending unit test.

### CODE REVIEW

**Comment by CalCraven** [comment]
I think the tests should go in `test_atom.py` and `test_top.py`. One that you can add a restraint to a site and parse it, and one that a topology with restraints will properly be id'd in gromacs. I think the best way to do that would be to make a `restrained_water_topology` in `base_test.py` for other engines to use, and then call that.

**Comment by daico007** [comment]
This PR includes the fix for the current failing CI tests.

**Comment by daico007** [comment]
All unit tests has been added.

File: gmso/utils/files/spce.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<ForceField version="0.0.1" name="SPC/E Water" combining_rule="geometric">
```
Comment (CalCraven) [inline gmso/utils/files/spce.xml:1]: I think this file already exists in the `utils/files/gmso_xmls/test_ff_styles` directory. I think you can probably delete it and source from there.

File: gmso/tests/base_test.py [hunk 2]
```diff
@@ -165,6 +165,24 @@ def _typed_topology(n_sites=100):
 
         return _typed_topology
 
+    @pytest.fixture
+    def spce_water(self):
+        spce_comp = mb.lib.molecules.water.WaterSPC()
+        spce_ff = ForceField(get_fn("spce.xml"))
```
Comment (CalCraven) [inline gmso/tests/base_test.py:171]:
```suggestion
        spce_ff = ForceField(get_fn("gmso_xmls/test_ffstyles/spce.xml"))
```

**Review by CalCraven — CHANGES_REQUESTED** [review]
Just remove the duplicate files and it should be good to go!

File: gmso/tests/base_test.py (hunk 2 above)
Comment (daico007) [reply gmso/tests/base_test.py]: Good catch! Done.

**Review by CalCraven — APPROVED** [review]
Once tests pass, lets merge it.

**Pull request merged**